Lib/gzip.py

```diff
@@ -95,7 +95,11 @@ def __init__(self, filename=None, mode=None,
         if filename is None:
             # Issue #13781: os.fdopen() creates a fileobj with a bogus name
             # attribute. Avoid saving this in the gzip header's filename field.
-            if hasattr(fileobj, 'name') and fileobj.name != '<fdopen>':
+            if (
```

Comment:
I suggest to backport the code from 3.x (with adding a special case for `'<fdopen>'`).